assignment/scripts/3-music-collection.js

```diff
@@ -1 +1,157 @@
+            artistArray.push(obj);
+        }
+    }
+    return artistArray;
+}
+
+// - Test the `findByArtist` function. Make sure to test with an artist you know is in the collection, as well as an artist you know is not in your collection. Check that for artists with multiple matches, all are found.
+
+console.log(findByArtist('Beyonce'));
+console.log(findByArtist('Taylor Swift'));
+console.log(findByArtist('Selena Gomez'));
+
+// > When testing your functions, write all tests in the JavaScript file!
+
+
+// ### Stretch goals
+
+// - Create a function called `search`. This function should:
+//   - Take an input parameter for a search criteria object. Create your solution based on a search object that has these properties:
+//   ```
+//   { artist: 'Ray Charles', year: 1957 }
+//   ```
+//   - The returned output from `search` should meet these requirements:
+//     - Return a new array of all items in the `collection` matching *all* of the search criteria.
+//     - If no results are found, return an empty array.
+//     - If there is no search object or an empty search object provided as input, then return all albums in the `collection`.
+
+
+let searchObject = {
+    artist: 'Ray Charles',
+    year: 1957
+}
+
+let taylorSwift = {
+    artist: 'Taylor Swift',
+    year: 2020
+}
+
+function search(searchCriteriaObject) {
+    if(arguments.length < 1) {
+        return collection;
+    }
+    let searchArray = [];
+    for(let obj of collection) {
+        if(searchCriteriaObject.artist === obj.artist && searchCriteriaObject.year === obj.year) {
+            searchArray.push(searchCriteriaObject)
+        }
+    }
+    return searchArray;
+}
+
+console.log(search(taylorSwift));
+console.log(search(searchObject));
+console.log(search());
+
+// - Add an array of `tracks` to your album objects. Each track should have a `name` and `duration`. You will need to update the functions to support this new property:
+//   - Update the `addToCollection` function to also take an input parameter for the array of tracks.
+//   - Update `search` to allow a `trackName` search criteria. 
+//     - IF the search object has a `trackName` property, only search for that, ignoring any `artist` or `year` properties.
+//   - Update the `showCollection` function to display the list of tracks for each album with its name and duration.
+// ```
+//     TITLE by ARTIST, published in YEAR:
+//     1. NAME: DURATION
+//     2. NAME: DURATION
+//     3. NAME: DURATION
+//     TITLE by ARTIST, published in YEAR:
+//     1. NAME: DURATION
+//     2. NAME: DURATION
+// ```
+
+// > Make sure to test all your code!
+
+//This is as far as I got, I am not sure if I am understand the intructions correctly
+
+function addToCollection(titleName, artistName, yearPublished, tracksArray) {
+    let newObject = {
+        title: titleName,
+        artist: artistName,
+        year: yearPublished,
+        tracks: [
+            {
+                trackName: tracksArray.trackName,
```

Comment:
Careful! The array of tracks does not have a property "trackName", the _individual track objects in the array do_. Since you named this function the same thing as your original `addToCollection`, this broken version is overwriting your working non-stretch version and throwing an error in your code.

I appreciate seeing your attempts at stretch problems, just be careful they don't conflict with your otherwise working code. One option here would be to name this function `addToCollectionStretch` or something.